src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileInfo.cs

```diff
@@ -35,7 +35,7 @@ public Stream CreateReadStream()
                 FileMode.Open,
                 FileAccess.Read,
                 FileShare.ReadWrite,
-                1024 * 64,
+                Math.Min((int)Length, 1024 * 64),
```

Comment:
What if `Length` is greater than `int.MaxValue`? I think it will overflow and would result in `-1` which would also throw. https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem/src/System/IO/FileStream.cs#L85

```c#
long lng = long.MaxValue;
var i = Math.Min((int)lng, 1);
// i == -1
```